### PR TITLE
Fix the CI because of a components::Bag and Time Zone conflict

### PR DESCRIPTION
**Pull request by gregtatum**
This was an issue with #617 and #601 and #598 colliding.

### CODE REVIEW

**Review by sffc — APPROVED** [review]
Looks like this encompasses #638 as well.  So I suggest we merge this ASAP and then close #638.

File: components/datetime/src/skeleton.rs [hunk 1]
```diff
@@ -375,11 +375,12 @@ pub fn create_best_pattern_for_fields<'a>(
...
     if !other.is_empty() {
-        // TODO(#418) - TimeZones
+        // These require "append items" support, see #586.
+        // TODO(#583) - TimeZones
         // TODO(#486) - Eras,
         // ... etc.
-        unimplemented!(
-            "There are no \"other\" fields supported, these need to be appended to the pattern."
+        eprintln!(
+            "There are no \"other\" fields supported, these need to be appended to the pattern. {:?}", other
```
Comment (zbraniecki) [inline components/datetime/src/skeleton.rs:383]: `question`: this doesn't seem to be related to the bustage and it is a functional change from a panic to a print to stderr. Why did you put it in this PR? I'm not sure if this is the right change tbh.

File: components/datetime/src/skeleton.rs [hunk 2]
```diff
@@ -722,6 +723,42 @@ mod test {
         };
     }
 
+    // TODO(#586) - Append items support needs to be added.
+    #[test]
```
Comment (gregtatum) [inline components/datetime/src/skeleton.rs:727]: @zbraniecki - I added this test since I didn't know what the behavior was here. It turns out the time zone fields are not even added by the components bag yet. I've found it helpful working incrementally to have tests like these that show when something is correctly implemented.

**Review by sffc — APPROVED** [review]

File: components/datetime/src/skeleton.rs (hunk 1 above)
Comment (gregtatum) [reply components/datetime/src/skeleton.rs:383]: I assumed this branch was going to be hit by my changes, but it's still not. See my latest commit for how I've addressed this question. I think the most technically correct solution would be to return a `BestSkeleton::MissingOrExtraFields` when a time zone is used, but I'm not sure it's worth the code diff to accomplish that. The components bag API is throwing away this information currently:

```
            match skeleton::create_best_pattern_for_fields(
                &self.date_time.skeletons,
                &self.date_time.length_patterns,
                &requested_fields,
            ) {
                skeleton::BestSkeleton::AllFieldsMatch(pattern)
                | skeleton::BestSkeleton::MissingOrExtraFields(pattern) => Some(pattern),
                skeleton::BestSkeleton::NoMatch => None,
            },
```

**Review by zbraniecki — APPROVED** [review]
lgtm!

File: components/datetime/src/skeleton.rs (hunk 1 above)
Comment (zbraniecki) [reply components/datetime/src/skeleton.rs:383]: yeah, I understand and am comfortable with WIP situation. I was uncertain about switching a panic to a println only. I don't think we should have any println or eprintln in the lib at this point.

Comment (sffc) [reply components/datetime/src/skeleton.rs:383]: Is there a follow-up issue for this?

I prefer keeping this PR small to fix the broken build.
